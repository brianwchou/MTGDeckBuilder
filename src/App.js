import React, { Component } from 'react';
import styled from 'styled-components';
import { fuzzySearchURL } from './URLs';
import './App.css';

class Card extends Component {
  constructor(props) {
    super(props)

    this.handleOnDragStart = this.handleOnDragStart.bind(this);
  }

  handleOnDragStart(e) {
    let {url} = this.props;
    e.dataTransfer.setData("text/plain", url);
  }

  render() {
    let {url} = this.props;
    return (
        <img draggable={true} 
          src={url}
          onDragStart={this.handleOnDragStart}      
        />
    )
  }
}

const cardBoxStyle = {
  border: 'solid black thin',
  width: '100vw',
  minHeight: '50vh',
}

class CardBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cards: []
    }

    this.drop = this.drop.bind(this);
    this.onDragOver = this.onDragOver.bind(this);
  }

  drop(e) {
    e.preventDefault();
    var item = e.dataTransfer.getData("text/plain");
    this.state.cards.push(item);
    this.setState({ ...this.state.cards });
  }

  onDragOver(e) {
    e.preventDefault();
    console.log('you are dragging over me');
  }

  render() {
    let cards = this.state.cards.map((url, index) => {
      return <Card url={url} key={index} />
    })

    return(
      <div style={cardBoxStyle} 
        onDrop={this.drop} 
        onDragOver={this.onDragOver}>
        {cards}
      </div>
    )
  }
}

class Search extends Component {
  constructor() {
    super();

    this.state = {
      cardURLs: [],
      searchCardName: "",
      insideCardBox: []
    };

    this.getCard = this.getCard.bind(this);
    this.clear = this.clear.bind(this);
    this.onCardNameChange = this.onCardNameChange.bind(this);
  }
  
  onCardNameChange(e) {
    this.setState({searchCardName: e.target.value})
  }

  async getCard(e) {
    e.preventDefault();


    var cardURL;

    var searchCardNameURL = fuzzySearchURL + this.state.searchCardName;
    console.log(searchCardNameURL);

    await fetch(searchCardNameURL)
    .then( response => {
      if (response.status === 200) {
        return response.json();
      } else {
        console.log("response", response.status)
        return null;
      }
    })
    .then((json) => {
      console.log(json)
      cardURL = (json !== null) ? json.image_uris.small : null;
    });

    console.log(cardURL);
    if (cardURL !== null ) {
      this.state.cardURLs.push(cardURL);
      this.setState({ ...this.state.cardURLs });
    }
  }

  clear() { this.setState({cardURLs: []}) }

  render() {
    var cards = this.state.cardURLs.map((url, index) => {
      return <Card key={index} url={url} />
    });

    return (
      <div>
        <form onSubmit={this.getCard} >
          <input type="text" onChange={this.onCardNameChange}/>
          <button type="submit"> submit </button>
          <button type="button" onClick={this.clear}> clear </button>
        </form>
        {cards}
      </div>
    )
  }
}

class App extends Component {
  render() {
    return (
      <div className="App">
        <Search />
        <CardBox>
        </CardBox>
      </div>
    );
  }
}

export default App;