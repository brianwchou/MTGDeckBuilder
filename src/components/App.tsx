import React from 'react';
import Container from '../Container';

const myName = "floopy poopty";

const App: React.FC = () => {
  return (
    <div>
      <Container name={myName}/>
    </div>
  );
}

export default App;
