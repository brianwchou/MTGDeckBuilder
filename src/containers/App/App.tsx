import { BrowserRouter, Route, Switch} from 'react-router-dom';
import './App.css';
import Navbar from '../../components/Navbar'
import NoMatch from '../../components/NoMatch';
import DeckBuilder from '../../containers/DeckBuilder';
import React from 'react';

const App = () => {
    return (
      <div>
        <Navbar />
        <BrowserRouter basename="/deck-builder-web/">
            <Switch>
                <Route exact path="/" component={DeckBuilder} />
                <Route component={NoMatch} />
            </Switch>
        </BrowserRouter>
      </div>
    )
}

export default App;