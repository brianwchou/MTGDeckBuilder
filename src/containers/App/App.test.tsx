import React from 'react';
import App from 'containers/App'
import { configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {

    test.only('renders correctly', () => {
        const wrapper = shallow(<App />)
        
        expect(wrapper).toMatchSnapshot();
    })
})