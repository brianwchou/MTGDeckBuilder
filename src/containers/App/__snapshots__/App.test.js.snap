// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar />,
        <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/home"
            />
            <Route
              exact={true}
              path="/"
            >
              <Redirect
                to="/home"
              />
            </Route>
            <Route
              component={[Function]}
            />
          </Switch>
        </BrowserRouter>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/home"
            />
            <Route
              exact={true}
              path="/"
            >
              <Redirect
                to="/home"
              />
            </Route>
            <Route
              component={[Function]}
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/home"
              />,
              <Route
                exact={true}
                path="/"
              >
                <Redirect
                  to="/home"
                />
              </Route>,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Redirect
                  to="/home"
                />,
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "to": "/home",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar />,
          <BrowserRouter>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/home"
              />
              <Route
                exact={true}
                path="/"
              >
                <Redirect
                  to="/home"
                />
              </Route>
              <Route
                component={[Function]}
              />
            </Switch>
          </BrowserRouter>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/home"
              />
              <Route
                exact={true}
                path="/"
              >
                <Redirect
                  to="/home"
                />
              </Route>
              <Route
                component={[Function]}
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/home"
                />,
                <Route
                  exact={true}
                  path="/"
                >
                  <Redirect
                    to="/home"
                  />
                </Route>,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/home",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Redirect
                    to="/home"
                  />,
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "to": "/home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
